plugins/inputs/tacacs/tacacs.go

```diff
@@ -0,0 +1,150 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package tacacs
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/nwaples/tacplus"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Tacacs struct {
+	Servers         []string        `toml:"servers"`
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	Secret          config.Secret   `toml:"secret"`
+	RemAddr         string          `toml:"request_ip"`
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	Log             telegraf.Logger `toml:"-"`
+	clients         []tacplus.Client
+	testAuthStart   tacplus.AuthenStart
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (t *Tacacs) SampleConfig() string {
+	return sampleConfig
+}
+
+func (t *Tacacs) Init() error {
+	if len(t.Servers) == 0 {
+		t.Servers = []string{"127.0.0.1:49"}
+	}
+	if net.ParseIP(t.RemAddr) == nil {
+		return fmt.Errorf("Invalid ip address provided for request_ip: %s", t.RemAddr)
+	}
+
+	for index, serverValue := range t.Servers {
+		t.clients = append(t.clients, tacplus.Client{})
```

Comment:
I think you can already make the size of your `clients` array the same as the `Servers` array, so no need to append each loop.